### DISCUSSION TITLE
Rules are not working for non-rendered components

### DISCUSSION THREAD

**Discussion by davidlruizc** [react-hook-form/react-hook-form#10156] [Q&A] [answered]
Hey guys! I'm having a problem using disclosures from `@headlessui/react` and a combination of useFormContext and useFieldArray. I created an example on codesandbox to elaborate it better.

https://codesandbox.io/s/disclosure-with-nested-form-vbcmdd?file=/src/App.js

The problem is the next one:
Form 1 rendered the information, and the rules run and send an error, everything OK with this, but for the other forms that aren't rendered yet do not count their rules and submit the information.
For the other forms to be validated by their rules the user need to open the disclosure section.

What I'm looking for is a way to keep the rules even though the Disclosure is closed or not.

I'll be grateful for the help 😄

**Comment by teamzz111** [comment]
Same issue.

**Comment by leapful** [comment, marked as answer]
@davidlruizc @teamzz111 

The root cause of this issue is because Disclosure component from @headless/react library always unmount its Panel content when if the content is not show yet.

Please use prop `unmount=false` on Panel component to keep content as hidden and still being registered with react-hook-form instead of unmount.

Working demo: https://codesandbox.io/s/disclosure-with-nested-form-forked-orbjy1?file=/src/App.js

**Reply by davidlruizc** [reply]
I see, I was looking information about the hidden elements but non of the solutions worked, it was about the Disclosure component. And now works perfect.

Thank you so much for the help.
